### PR TITLE
[react-native] Added autoCompleteType to TextInputAndroidProps

### PR DESCRIPTION
**Pull request by danilobuerger**
See https://github.com/facebook/react-native/pull/21575

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/pull/21575
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-native/index.d.ts [hunk 1]
```diff
@@ -1107,6 +1107,43 @@ export interface TextInputIOSProps {
...
+     * - `tel`
+     * - `street-address`
+     * - `postal-code`
+     * - `cc-number`
+     * - `cc-csc`
+     * - `cc-exp`
+     * - `cc-exp-month`
+     * - `cc-exp-year`
+     * - `off`
+     */
```
Comment (alloy) [inline types/react-native/index.d.ts]: Is it really needed to list all of these in the comment? That will require keeping them up-to-date.

**Review by alloy — COMMENTED** [review]
Could you add a test that uses (some of) these?

File: types/react-native/index.d.ts (hunk 1 above)
Comment (danilobuerger) [reply types/react-native/index.d.ts]: The documentation is provided by https://github.com/facebook/react-native/blob/master/Libraries/Components/TextInput/TextInput.js#L438-L461 
I did not modify it, just copy & paste.

**Comment by danilobuerger** [comment]
@alloy added test

**Comment by alloy** [comment]
Ok, works for me then 👍

**Pull request merged**